slider/beatmap.py

```diff
@@ -660,6 +661,31 @@ def tick_points(self):
             ),
         )
 
+    @lazyval
+    def true_tick_points(self):
+        tick_points = self.tick_points
+        # curve() takes in a percentage of how far along we want the point.
+        # Take away the offset from the total length of the slider to get
+        # the percentage of the slider we want the point at.
+        true_end_time = (
+            self.end_time - self.LEGACY_LAST_TICK_OFFSET
+        )
+
+        # keep in mind to check for negative values here, e.g. at least
+        # have the duration be 0
+        legacy_duration = max(
+            true_end_time - self.time, timedelta(milliseconds=0)
+        )
+        real_duration = self.end_time - self.time
+
+        ratio = legacy_duration / real_duration
+        curve_point = int(self.length * ratio)
+        pos = self.curve(curve_point / self.length)
+
+        tick_points[-1] = Point(pos.x, pos.y, true_end_time)
```

Comment:
this is a dangerous assignment since it might change the cached value of `self.tick_points`. It doesn't seem to in my testing and I'm frankly not 100% sure why. I'd prefer to avoid a deepcopy here if possible though, so I'm ok letting this slide.